modules/virtualization/microvm/appvm.nix

```diff
@@ -0,0 +1,176 @@
+
+          nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
+          nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
+
+          networking = {
+            enableIPv6 = false;
+            interfaces.ethint0.useDHCP = false;
+            firewall.allowedTCPPorts = [22];
+            firewall.allowedUDPPorts = [67];
+            useNetworkd = true;
+          };
+
+          environment.systemPackages = [
+            pkgs.waypipe
+          ];
+
+          microvm = {
+            mem = vm.ramMb;
+            vcpu = vm.cores;
+            hypervisor = "qemu";
+            qemu.bios.enable = true;
+            storeDiskType = "squashfs";
+            interfaces = [
+              {
+                type = "tap";
+                id = hostname;
+                mac = vm.macAddress;
+              }
+            ];
+          };
+
+          networking.nat = {
+            enable = true;
+            internalInterfaces = ["ethint0"];
+          };
+
+          # Set internal network's interface name to ethint0
+          systemd.network.links."10-ethint0" = {
+            matchConfig.PermanentMACAddress = vm.macAddress;
+            linkConfig.Name = "ethint0";
+          };
+
+          systemd.network = {
+            enable = true;
+            networks."10-ethint0" = {
+              matchConfig.MACAddress = vm.macAddress;
+              addresses = [
+                {
+                  # IP-address for debugging subnet
+                  addressConfig.Address = vm.ipAddress;
+                }
+              ];
+              routes = [
+                {routeConfig.Gateway = "192.168.101.1";}
+              ];
+              linkConfig.RequiredForOnline = "routable";
+              linkConfig.ActivationPolicy = "always-up";
+            };
+          };
+
+          imports = import ../../module-list.nix;
+        })
+      ];
+    };
+  in {
+    autostart = true;
+    config = appvmConfiguration // {imports = appvmConfiguration.imports ++ cfg.extraModules ++ [{environment.systemPackages = vm.packages;}];};
+    specialArgs = {inherit lib;};
+  };
+in {
+  options.ghaf.virtualization.microvm.appvm = with lib; {
+    enable = lib.mkEnableOption "appvm";
+    vms = with types;
+      mkOption {
+        description = ''
+          List of AppVMs to be created
+        '';
+        type = lib.types.listOf (submodule {
+          options = {
+            name = mkOption {
+              description = ''
+                Name of the AppVM
+              '';
+              type = str;
+            };
+            packages = mkOption {
+              description = ''
+                Packages that are included into the AppVM
+              '';
+              type = types.listOf package;
+              default = [];
+            };
+            ipAddress = mkOption {
+              description = ''
+                AppVM's IP address in the inter-vm network
```

Comment:
This addresses the VMs part of my earlier comment https://github.com/tiiuae/ghaf/pull/243/files#r1302862794
Ghaf inter-vm network subnet/mask comment still applies - for example when one would like to change the subnet from `192.168.101.0` to say `10.0.0.0` (or something else)

Reply:
SP-2839 is created to track this